assets/js/translator.js

```diff
@@ -188,6 +217,90 @@ if(modeEnabled('translation')) {
             $('a#fileDownload').fadeOut('fast');
         });
 
+        $('#translatedText').css('height', $('#originalText').css('height'));
+        $('#suggestCloseBtn').click(function() {
+            $('#suggestedWordInput').val('');
+            grecaptcha.reset();
+        });
+        $('#suggestBtn').click(function() {
+            var fromWord = $('#suggestionTargetWord').html();
+            var toWord = $('#suggestedWordInput').val();
+            var recaptchaResponse = grecaptcha.getResponse();
+
+            if(toWord.length === 0) {
+                $('#suggestedWordInput').tooltip('destroy');
+                $('#suggestedWordInput').tooltip({
+                    'title': 'Suggestion cannot be empty.',
+                    'trigger': 'manual',
+                    'placement': 'bottom'
+                });
+                $('#suggestedWordInput').tooltip('show')
+                setTimeout(function() {
+                    $('#suggestedWordInput').tooltip('hide');
+                    $('#suggestedWordInput').tooltip('destroy');
+                }, 3000);
+
+                return;
+            }
+
+            // Obtaining context, (± config.SUGGESTIONS.context_wrap) words
+            // fallback to complete text if this fails.
+            var hashedWord = fromWord.hashCode() + fromWord + fromWord.hashCode();
+            $('#wordGettingSuggested').text(hashedWord);
+
+            var splitted = $('#translatedText').text().split(' ');
+            $('#wordGettingSuggested').text(fromWord)
+
+            var targetIndex = splitted.indexOf(hashedWord);
+            var wrapLength = parseInt(config.SUGGESTIONS.context_wrap);
+            var begin, end;
+            begin  = (targetIndex > wrapLength) ? (targetIndex - wrapLength) : 0;
+            end = (splitted.length - targetIndex - 1 > wrapLength) ? (targetIndex + wrapLength + 1) : splitted.length;
+            
+            var context = splitted.slice(begin, end).join(' ').replace(hashedWord, fromWord);
+            if (!context) {
+                context = $('#translatedText').attr('pristineText');
+            }
+
+            $.ajax({
+                url: config.APY_URL + '/suggest',
+                type: 'POST',
+                beforeSend: ajaxSend,
+                data: {
+                    'langpair': curSrcLang + '|' + curDstLang,
+                    'word': fromWord,
+                    'newWord': toWord,
+                    'context': context,
+                    'g-recaptcha-response': recaptchaResponse
+                },
+                success: function (data) {
+                    $('#suggestedWordInput').tooltip('hide');
```

Comment:
Do we really need to hide it before destroying or does this actually do something? o.o